### PR TITLE
Feature: Assignment Submission

### LINKED ISSUE #167: Fix Datetime Formatting for Assignment Deadline
The simple_form for one is kinda shitty and it's ugly af but since this isn't a consumer facing feature this is low priority.

### LINKED ISSUE #175: Assignment Submssion

### PR DESCRIPTION
**Pull request by chenhoward**
## About
Since we have a website for checking in we might as well have the ability to all our grading on it.  Now they can submit their homework through our site.  I fixed some styling for the dates as well.

## Issues
Fix #175
Fix #167

### CODE REVIEW

File: app/views/assignments/show.html.slim [hunk 1]
```diff
@@ -1,8 +1,13 @@
 .container
   h1 = @assignment.name
-  h2 = "Due #{@assignment.deadline}"
+  h2 = "Due #{@assignment.formatted_deadline}"
   h2 = "Points: #{@assignment.points}"
-  a href= @assignment.link Link
+  a href= @assignment.link Link to Assignment
+  - if current_user && current_user.is_current_student?
```
Comment (DrakeW) [inline app/views/assignments/show.html.slim]: this part can be simplified as  ```current_user.try(:is_current_student?)``` :)

File: app/models/assignment_submission.rb [hunk 2]
```diff
@@ -10,9 +10,29 @@
...
+  validates :user, presence: true
+  validates :grader, presence: true, if: :is_graded?
+
+  private
+
+  def grader_is_staff?
+    if grader
+      return grader.is_staff_for_semester? assignment.semester
+    end
+    false
```
Comment (DrakeW) [inline app/models/assignment_submission.rb]: can be simplified as ```grader.nil? ? false : grader.is_staff_for_semester?(assignment.semester)```

File: app/controllers/assignments_controller.rb [hunk 3]
```diff
@@ -5,6 +5,12 @@ def index
 
   def show
     @assignment = Assignment.find(params[:id])
+    users_assignments = @assignment.assignment_submissions.where(user: current_user)
+    @assignment_submission = users_assignments.first_or_initialize
+    @button_text = "Submit Assignment"
+    if @assignment_submission.id
+      @button_text = "Update Submission Link"
+    end
```
Comment (DrakeW) [inline app/controllers/assignments_controller.rb]: I'm just a fan of one liner haha...
```@button_text = @assignment_submission.id.nil? ? "Update Submission Link" : "Submit Assignment"```

File: app/views/assignments/show.html.slim (hunk 1 above)
Comment (chenhoward) [inline app/views/assignments/show.html.slim]: Ohh that's useful

File: app/controllers/assignments_controller.rb (hunk 3 above)
Comment (chenhoward) [inline app/controllers/assignments_controller.rb]: I actually hate the ternary lol

File: app/models/assignment_submission.rb (hunk 2 above)
Comment (chenhoward) [inline app/models/assignment_submission.rb]: Hate the ternary lul

File: app/controllers/assignments_controller.rb (hunk 3 above)
Comment (DrakeW) [inline app/controllers/assignments_controller.rb]: :cry: :cry: :cry:

Comment (chenhoward) [inline app/controllers/assignments_controller.rb]: I will do it though since it is the what airbnb says LOL

**Comment by chenhoward** [comment]
@DrakeW did it coach!

**Comment by DrakeW** [comment]
@chenhoward haha looks good :+1:

**Pull request merged**